### PR TITLE
Generate pretty state diffs

### LINKED ISSUE #1561: Generate pretty state-diffs.
**Issue summary**

The statediff library should attempt to parse IPLD data into Rust structures and pretty-print them. If that fails, we can default to printing the IPLD as JSON.

**Other information and links**

See `utils/statediff/src/lib.rs`

### PR DESCRIPTION
**Pull request by connormullett**
**Summary of changes**
Changes introduced in this pull request:
- Using state structs, attempt to parse the rest of the IPLD data as rust data structures before defaulting to IPLD JSON.

**Reference issue to close (if applicable)**

Closes #1561 

**Other information and links**

### CODE REVIEW

File: utils/statediff/src/lib.rs [hunk 1]
```diff
@@ -158,6 +160,133 @@ pub struct MinerState {
     pub deadline_cron_active: bool,
```
Comment (LesnyRumcajs) [inline utils/statediff/src/lib.rs:160]: Could we import those structures from `builtin-actors` instead? If some additional traits are needed, we could make a PR there.

**Comment by lemmih** [comment]
@connormullett Remember to fix the spellcheck failures.

**Comment by connormullett** [comment]
> @connormullett Remember to fix the spellcheck failures.

Removing the structs per @LesnyRumcajs mention and just making a PR in the builtin-actors crate will remove the spellcheck issues. Just have to open that PR

**Comment by LesnyRumcajs** [comment]
> > @connormullett Remember to fix the spellcheck failures.
> 
> Removing the structs per @LesnyRumcajs mention and just making a PR in the builtin-actors crate will remove the spellcheck issues. Just have to open that PR

If the PR is accepted, it may take some time to have it packaged and released. Perhaps to not wait, let's create a separate issue to use imports instead of copying and let's merge this now. What do you think?

File: utils/statediff/src/lib.rs [hunk 2]
```diff
@@ -158,6 +160,133 @@ pub struct MinerState {
...
 }
 
+#[derive(Clone, Default, Serialize_tuple, Deserialize_tuple, Debug)]
```
Comment (LesnyRumcajs) [inline utils/statediff/src/lib.rs:163]: To my understanding, this won't (and should not) be used outside the statediff. How about making those private or `pub(crate)` so they won't play tricks on others? In other words, I expect my completion to suggest me to import e.g. the `MarketState` from `builtin-actors` and not from here.

Comment (connormullett) [reply utils/statediff/src/lib.rs:163]: Good point, I'll fix up the access throughout the crate to be sure

**Review by LesnyRumcajs — APPROVED** [review]
looks good!

**Review by lemmih — APPROVED** [review]

**Pull request merged**